### DISCUSSION TITLE
Java connect mysql unit

### DISCUSSION THREAD

**Discussion by qingguox** [ml-for-good/automl#107] [General]
Java链接mysql的工具讨论：
1、原生jdbc，比较推荐，一个dao就可以方便的做prod、test环境分离. 目前项目看起来逻辑是不太复杂了，单表操作即可。不用复杂的联表查询，直接在内存中处理业务逻辑即可。
2、mybatis，太重了， 不太方便做多数据源，因为prod和test表结构完全一样，只是mapper目录变了(一个表两个mapper),  维护起来比较麻烦。

所以想讨论一下这个项目java链接mysql的工具是采用哪个，或者推荐一个更好用的工具。
@free-y @Kaychang @all

**Comment by qingguox** [comment]
我个人建议使用第一种。
如果大多数小伙伴采用第二种的话，可以说一下环境隔离时（多个数据源比如 线上和测试环境）如何方便的处理。

**Reply by qingguox** [reply]
+ mysql版本，如果是8，麻烦把部署的mysql也改成8.

**Reply by free-y** [reply]
需要修改mysql版本的话，直接提交一个PR即可。关于DAO的变更，我这边觉得，可以提PR以后群内投票，或者我直接在群里发一个投票

**Reply by qingguox** [reply]
可以，你发起一下吧
然后咱们的版本是需要用8.0吗？
感觉没啥必要吧
5.7 和5.8够用了
主要是两者驱动不一样，早点定下来就能开发了

**Reply by free-y** [reply]
现在myql我看已经是用的5.7的镜像

**Reply by qingguox** [reply]
那就用mysql5.7 我觉得可以，比较稳定。
